Does not build with `NO_DEBUG_BUFFER` removed

Hi Sergi!
  Unfortunately I'm getting some linker errors when commenting out `NO_DEBUG_BUFFER` (if I don't chenge include/debug.h, then it works fine):

```
/opt/devkitpro/devkitARM/bin/arm-none-eabi-gcc -mcpu=arm926ej-s -mthumb -mthumb-interwork -mbig-endian -nostartfiles -Wl,-T,link.ld,-Map,MLOAD.map -Wl,--gc-sections -Wl,-static source/debug.o source/detect.o source/elf.o source/gecko.o source/gpio.o source/main.o source/patches.o source/start.o source/swi.o source/swi_asm.o source/tid.o cios-lib/cios-lib.a -lgcc -o MLOAD.elf
/opt/devkitpro/devkitARM/bin/../lib/gcc/arm-none-eabi/14.2.0/../../../../arm-none-eabi/bin/ld: MLOAD.elf section `.text' will not fit in region `exe'
/opt/devkitpro/devkitARM/bin/../lib/gcc/arm-none-eabi/14.2.0/../../../../arm-none-eabi/bin/ld: section .data LMA [136d2000,136d2257] overlaps section .text LMA [136d0060,136d2017]
/opt/devkitpro/devkitARM/bin/../lib/gcc/arm-none-eabi/14.2.0/../../../../arm-none-eabi/bin/ld: region `exe' overflowed by 24 bytes
collect2: error: ld returned 1 exit status
make: *** [Makefile:63: MLOAD.elf] Error 1
```

I guess it's the values in `link.ld` that make it fail. I see that changing the values in it like this
```diff
diff --git a/link.ld b/link.ld
index 25e05d8..1dd9f2b 100644
--- a/link.ld
+++ b/link.ld
@@ -5,13 +5,13 @@ ENTRY(_start)
 /* Sections area */
 MEMORY {
        table           : ORIGIN = 0x0, LENGTH = 0x4000
-       exe(rwx)        : ORIGIN = 0x136d0000, LENGTH = 0x2000
-       ram(rw)         : ORIGIN = 0x136d2000, LENGTH = 0x9000
+       exe(rwx)        : ORIGIN = 0x136d0000, LENGTH = 0x2400^M
+       ram(rw)         : ORIGIN = 0x136d2400, LENGTH = 0x9000^M
        mem(rw)         : ORIGIN = 0x13700000, LENGTH = 0x80000
 }
 
 __exe_start_phys__             = 0x136d0000;
-__ram_start_phys__             = 0x136d2000;
+__ram_start_phys__             = 0x136d2400;^M
 __mem_start_phys__             = 0x13700000;
 

```

make the project build, but I'm not familiar with this file and I don't know if these changes are good. Please let me know, I will than make a PR. Or if needed, we can add two versions of this map file, and select the proper one from the Makefile (if we make `DEBUG_BUFFER` be a make flag).

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -24,7 +24,7 @@ jobs:
       run: make -j 2
 
     - name: Upload artifacts
-      uses: actions/upload-artifact@v3
+      uses: actions/upload-artifact@v4
       with:
         name: MLOAD
         if-no-files-found: error

--- a/link.ld
+++ b/link.ld
@@ -5,13 +5,13 @@ ENTRY(_start)
 /* Sections area */
 MEMORY {
 	table		: ORIGIN = 0x0, LENGTH = 0x4000
-	exe(rwx)	: ORIGIN = 0x136d0000, LENGTH = 0x2000
-	ram(rw)		: ORIGIN = 0x136d2000, LENGTH = 0x9000
+	exe(rwx)	: ORIGIN = 0x136d0000, LENGTH = 0x2400
+	ram(rw)		: ORIGIN = 0x136d2400, LENGTH = 0x9000
 	mem(rw)		: ORIGIN = 0x13700000, LENGTH = 0x80000
 }
 
 __exe_start_phys__		= 0x136d0000;
-__ram_start_phys__		= 0x136d2000;
+__ram_start_phys__		= 0x136d2400;
 __mem_start_phys__		= 0x13700000;
 
 

